File: app/src/main/java/alex/com/githubchecker/components/github/list/PRViewHolder.java
package alex.com.githubchecker.components.github.list;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.jakewharton.rxbinding2.view.RxView;

import alex.com.githubchecker.R;
import alex.com.githubchecker.components.app.api.model.PullRequest;
import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.subjects.PublishSubject;

/**
 * Created by Alex on 11/13/2017.
 */

class PRViewHolder extends RecyclerView.ViewHolder {

    private View view;
    private PublishSubject<Integer> selectedPRSubject;

    @BindView(R.id.title) TextView titleTv;
    @BindView(R.id.status) TextView statusTv;

    private Integer _pullRequestID;

    PRViewHolder(View itemView, PublishSubject<Integer> selectedPRSubject) {
        super(itemView);
        this.view = itemView;
        this.selectedPRSubject = selectedPRSubject;
        ButterKnife.bind(this, view);

        RxView.clicks(view).subscribe(click -> {selectedPRSubject.onNext(_pullRequestID);});
    }

    void bind(PullRequest pullRequest) {

        _pullRequestID = pullRequest.getId();

        String userLogin = pullRequest.getUserLogin();
        Integer number = pullRequest.getNumber();
        String createdAt = pullRequest.getCreatedAt();
        String status = view.getContext().getResources().getString(R.string.pr_content, number, createdAt, userLogin);

        titleTv.setText(pullRequest.getTitle());
        statusTv.setText(status);
    }

}

File: app/src/main/java/alex/com/githubchecker/components/github/list/PRViewHolder.kt
package alex.com.githubchecker.components.github.list

import alex.com.githubchecker.R
import alex.com.githubchecker.models.room.dao.NestedPullRequest
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.jakewharton.rxbinding2.view.RxView
import io.reactivex.subjects.PublishSubject
import kotlinx.android.synthetic.main.item_pr.view.status
import kotlinx.android.synthetic.main.item_pr.view.title

/**
 * Created by Alex on 11/13/2017.
 */

class PRViewHolder(private val containerView: View, selectedPRSubject: PublishSubject<Int>) : RecyclerView.ViewHolder(containerView) {

    private var _pullRequestNumber: Int? = null

    init {
        RxView.clicks(containerView).subscribe { click -> selectedPRSubject.onNext(_pullRequestNumber!!) }
    }

    fun bind(nestedPullRequest: NestedPullRequest) {

        _pullRequestNumber = nestedPullRequest.pullRequest.number

        val userLogin = nestedPullRequest.user.login
        val number = nestedPullRequest.pullRequest.number
        val createdAt = nestedPullRequest.pullRequest.created_at
        val status = containerView.context.resources.getString(R.string.pr_content, number, createdAt, userLogin)

        containerView.title.text = nestedPullRequest.pullRequest.title
        containerView.status.text = status
    }
}

